src/sas/qtgui/Perspectives/Fitting/PolydispersityWidget.py

```diff
@@ -0,0 +1,512 @@
+            # Is the parameter checked for fitting?
+            value = item.checkState()
+
+            if value == QtCore.Qt.Checked:
+                self.poly_params_to_fit.append(parameter_name_w)
+            else:
+                if parameter_name_w in self.poly_params_to_fit:
+                    self.poly_params_to_fit.remove(parameter_name_w)
+            self.cmdFitSignal.emit()
+            # self.updateUndo()
+
+        elif model_column in [delegate.poly_min, delegate.poly_max]:
+            try:
+                value = GuiUtils.toDouble(item.text())
+            except TypeError:
+                # Can't be converted properly, bring back the old value and exit
+                return
+
+            current_details = self.logic.kernel_module.details[parameter_name_w]
+            if self.has_poly_error_column:
+                # err column changes the indexing
+                current_details[model_column-2] = value
+            else:
+                current_details[model_column-1] = value
+
+        elif model_column == delegate.poly_function:
+            # name of the function - just pass
+            pass
+
+        else:
+            try:
+                value = GuiUtils.toDouble(item.text())
+            except TypeError:
+                # Can't be converted properly, bring back the old value and exit
+                return
+
+            # Update the sasmodel
+            # PD[ratio] -> width, npts -> npts, nsigs -> nsigmas
+            if model_column not in delegate.columnDict():
+                return
+            self.poly_params[parameter_name_w] = value
+            self.logic.kernel_module.setParam(parameter_name_w, value)
+
+            # Update plot
+            self.updateDataSignal.emit()
+
+    def checkedListFromModel(self) -> List[str]:
+        """
+        Returns list of checked parameters for given model
+        """
+        def isChecked(row: int) -> bool:
+            return self.poly_model.item(row, 0).checkState() == QtCore.Qt.Checked
+
+        return [self.polyNameToParam(str(self.poly_model.item(row_index, 0).text()))
+                for row_index in range(self.poly_model.rowCount())
+                if isChecked(row_index)]
+
+    def togglePoly(self, isChecked: bool = True) -> None:
+        """
+        Toggle polydispersity
+        """
+        self.isActive = isChecked
+        # Set sasmodel polydispersity to 0 if polydispersity is unchecked, if not use Qmodel values
+        if self.poly_model.rowCount() > 0:
+            for key, value in self.poly_params.items():
+                if key[-6:] == '.width':
+                    self.logic.kernel_module.setParam(key, (value if isChecked else 0))
+
+    def updateModel(self, model: Optional[Any] = None) -> None:
+        # add polydisperse parameters if asked
+        if self.isActive and self.poly_model.rowCount() > 0:
+            for key, value in self.poly_params.items():
+                model.setParam(key, value)
```

Comment:
After switching from `core_multi_shell` with polydispersity enabled to `sphere`, I get the following error when I try to calculate the model. This does not happen in v6.0:

> ValueError: Model does not contain parameter thickness1.width

Reply:
All issues addressed (+ some more found out during testing)